OctoWS2811.cpp

```diff
@@ -70,224 +154,391 @@ OctoWS2811::OctoWS2811(uint32_t numPerStrip, void *frameBuf, void *drawBuf, uint
+
+    // pin 15 triggers DMA(port C) on falling edge of low duty waveform
+    // pin 15 and 16 must be connected by the user: 16 is output, 15 is input
+    pinMode(15, INPUT);
+    CORE_PIN15_CONFIG = PORT_PCR_IRQC(2)|PORT_PCR_MUX(1);
+
+    // pin 4 triggers DMA(port A) on falling edge of high duty waveform
+    CORE_PIN4_CONFIG = PORT_PCR_IRQC(2)|PORT_PCR_MUX(3);
+
+#elif defined(KINETISL)
+    // on Teensy-LC, use timer DMA, not pin DMA
+    //Serial1.println(FTM2_C0SC, HEX);
+    //FTM2_C0SC = 0xA9;
+    //FTM2_C0SC = 0xA9;
+    //uint32_t t = FTM2_C0SC;
+    //FTM2_C0SC = 0xA9;
+    //Serial1.println(t, HEX);
+    CORE_PIN3_CONFIG = 0;
+    CORE_PIN4_CONFIG = 0;
+    //FTM2_C0SC = 0;
+    //FTM2_C1SC = 0;
+    //while (FTM2_C0SC) ;
+    //while (FTM2_C1SC) ;
+    //FTM2_C0SC = 0x99;
+    //FTM2_C1SC = 0x99;
+
+    //MCM_PLACR |= MCM_PLACR_ARB;
+
+#endif
+
+    // DMA channel #1 sets WS2811 high at the beginning of each cycle
+    dma1.source(ones);
+    dma1.destination(GPIOD_PSOR);
+    dma1.transferSize(1);
+    dma1.transferCount(bufsize);
+    dma1.disableOnCompletion();
+
+    // DMA channel #2 writes the pixel data at 20% of the cycle
+    dma2.sourceBuffer((uint8_t *)frameBuffer, bufsize);
+    dma2.destination(GPIOD_PDOR);
+    dma2.transferSize(1);
+    dma2.transferCount(bufsize);
+    dma2.disableOnCompletion();
+
+    // DMA channel #3 clear all the pins low at 48% of the cycle
+    dma3.source(ones);
+    dma3.destination(GPIOD_PCOR);
+    dma3.transferSize(1);
+    dma3.transferCount(bufsize);
+    dma3.disableOnCompletion();
+    dma3.interruptAtCompletion();
 
 #ifdef __MK20DX256__
-	MCM_CR = MCM_CR_SRAMLAP(1) | MCM_CR_SRAMUAP(0);
-	AXBS_PRS0 = 0x1032;
+    MCM_CR = MCM_CR_SRAMLAP(1) | MCM_CR_SRAMUAP(0);
+    AXBS_PRS0 = 0x1032;
 #endif
 
-	// route the edge detect interrupts to trigger the 3 channels
-	dma1.triggerAtHardwareEvent(DMAMUX_SOURCE_PORTB);
-	dma2.triggerAtHardwareEvent(DMAMUX_SOURCE_PORTC);
-	dma3.triggerAtHardwareEvent(DMAMUX_SOURCE_PORTA);
+#if defined(KINETISK)
+    // route the edge detect interrupts to trigger the 3 channels
+    dma1.triggerAtHardwareEvent(DMAMUX_SOURCE_PORTB);
+    dma2.triggerAtHardwareEvent(DMAMUX_SOURCE_PORTC);
+    dma3.triggerAtHardwareEvent(DMAMUX_SOURCE_PORTA);
+#elif defined(KINETISL)
+    // route the timer interrupts to trigger the 3 channels
+    dma1.triggerAtHardwareEvent(DMAMUX_SOURCE_FTM2_OV);
+    dma2.triggerAtHardwareEvent(DMAMUX_SOURCE_FTM2_CH0);
+    dma3.triggerAtHardwareEvent(DMAMUX_SOURCE_FTM2_CH1);
+#endif
 
-	// enable a done interrupts when channel #3 completes
-	dma3.attachInterrupt(isr);
-	//pinMode(1, OUTPUT); // testing: oscilloscope trigger
+    // enable a done interrupts when channel #3 completes
+    dma3.attachInterrupt(isr);
+    //pinMode(9, OUTPUT); // testing: oscilloscope trigger
 }
 
 void OctoWS2811::isr(void)
 {
-	dma3.clearInterrupt();
-	update_completed_at = micros();
-	update_in_progress = 0;
+    //Serial1.print(".");
```

Comment:
Looked at this and I did not add it.  I also see items added for KINETISL that I did not add.  It must have been added in PJRC version I merged with.  Do you want me to back out these changes as well?  Or is there a different repository that I should be merging with or that is closer to that used by PJRC?

What do you want me to do?